src/integrations/intercom/services/conversations-getter.js

```diff
@@ -51,56 +40,62 @@ function ConversationsGetter(Implementation, params, opts, collectionName) {
       });
   }
 
-  this.perform = () => {
-    model = Implementation.getModels()[collectionName];
-
-    return Implementation.Intercom.getCustomer(model, params.recordId)
-      .then((customer) => intercom.conversations
-        .list({
-          email: customer.email,
-          type: 'user',
-          display_as: 'plaintext',
-        })
-        .then((response) => {
-          const { conversations } = response.body;
-
-          if (response.body.pages.next) {
-            return fetchPages(response.body.pages, conversations);
-          }
-          return conversations;
-        })
-        .then((conversations) => [conversations.length,
-          conversations.slice(getSkip(), getSkip() + getLimit())])
-        .spread((count, conversations) => intercom.admins.list()
-          .then((response) => {
-            const { admins } = response.body;
-
-            return P
-              .map(conversations, (conversation) => {
-                if (conversation.assignee && conversation.assignee.type === 'admin') {
-                  const adminId = parseInt(conversation.assignee.id, 10);
-                  const admin = _.find(admins, { id: adminId });
-
-                  conversation.assignee = admin;
-                }
-
-                if (opts.integrations.intercom.apiKey) {
-                  conversation.link = getLink(conversation);
-                }
-
-                return conversation;
-              })
-              .then((conversationsFormatted) => [count, conversationsFormatted]);
-          }))
-        .catch((error) => {
-          try {
-            logger.error(`Cannot access to Intercom conversations: ${error.body.errors[0].message}`);
-          } catch (tryError) {
-            logger.error('Cannot access to Intercom conversations: ', error);
-          }
+  // NOTICE: Forcing the usage of bluebird to avoid issues with the spread on route file
+  this.perform = () => P.resolve(
+    ContactGetter
+      .getContact(intercom, Implementation, mappingValue, params.recordId)
+      .then((contact) => {
+        if (!contact || !contact.body || !contact.body.data || !contact.body.data[0]) {
+          logger.error('Cannot access to Intercom conversations: No intercom contact matches the given key');
           return [0, []];
```

Comment:
[Error handling] I don't throw errors here anymore as it ended up in an `unexpected error` instead of just a not found.